### PR TITLE
[Data Exfiltration Detection] Update files to add Serverless support to the package

### PR DESCRIPTION
**Pull request by sodhikirti07**
## What does this PR do?
This PR adds serverless support to the `Data Exfiltration Detection` advanced analytics package. 

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## How to test this PR locally

- Follow the steps to spin a Serverless project using `elastic-package`
- Make sure the format spec for `manifest.yml` is `3.0.0`
- Test if all the assets i.e. anomaly detection jobs, detection rules and dashboard are getting installed successfully.

## Screenshots

- Package Installation:
image

- Anomaly detection jobs installed and started:
image

image

- Security rules are enabled:
image

- Dashboard is installed properly:
image

### CODE REVIEW

File: packages/ded/changelog.yml [hunk 1]
```diff
@@ -1,3 +1,8 @@
+- version: "1.2.0"
```
Comment (ajosh0504) [inline packages/ded/changelog.yml:1]:
```suggestion
- version: "1.1.0"
```

File: packages/ded/manifest.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
-format_version: 2.2.0
+format_version: 3.0.0
 name: ded
 title: "Data Exfiltration Detection"
-version: 1.0.3
+version: 1.2.0
```
Comment (ajosh0504) [inline packages/ded/manifest.yml:4]:
```suggestion
version: 1.1.0
```

File: packages/ded/manifest.yml [hunk 3]
```diff
@@ -11,9 +11,12 @@ categories:
   - network
   - advanced_analytics_ueba
 conditions:
-  kibana.version: "^8.5.0"
+  kibana:
+    version: '^8.5.0'
```
[the remark this answers is not in the document]
Comment (sodhikirti07) [reply packages/ded/manifest.yml:15]: There's no specification for using a minimum stack version with the packages in the v3 spec.

[the remark this answers is not in the document]
Comment (sodhikirti07) [reply packages/ded/manifest.yml:15]: I checked with the serverless team and they confirm that we don't require a minimum Kibana version for serverless packages and they'll be based on spec versions. This condition is specifically kept for on-premise or stateful clusters.

**Review by ajosh0504 — APPROVED** [review]
LGTM!

**Pull request merged**